File: mobile-app/lib/models/produit.dart
class Produit {
  final int id;
  final String nom;
  final String description;
  final double prix;
  final String? imageUrl;
  final Map<String, dynamic>? categorie;

  Produit({
    required this.id,
    required this.nom,
    required this.description,
    required this.prix,
    this.imageUrl,
    this.categorie,
  });

  factory Produit.fromJson(Map<String, dynamic> json) {
    return Produit(
      id: json['id'],
      nom: json['nom'],
      description: json['description'] ?? '',
      prix: (json['prix'] as num).toDouble(),
      imageUrl: json['imageUrl'],
      categorie: json['categorie'],
    );
  }

  String get prixFormatted => '${prix.toStringAsFixed(2)} DT';
}

File: produits-service/src/main/java/com/boutique/produits/entity/Produit.java
package com.boutique.produits.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "produits")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Produit {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 200)
    private String nom;
    
    @Column(length = 1000)
    private String description;
    
    @Column(nullable = false)
    private Double prix;
    
    private String imageUrl;
    
    @ManyToOne
    @JoinColumn(name = "categorie_id")
    private Categorie categorie;
    
    public Produit(String nom, String description, double prix, Categorie categorie) {
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.categorie = categorie;
    }
}
